tests/pytests/test_multibyte_char_terms.py

```diff
@@ -0,0 +1,227 @@
+
+        # Search consonants
+        expected = [2, 'test:consonantsU', 'test:consonantsL']
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', '@t:БВГДЗКЛМНПРСТФХ', 'NOCONTENT', 'SORTBY', 't')
+        env.assertEqual(res, expected, message=f'Dialect: {dialect}')
+
+        # Search soft consonants
+        expected = [2, 'test:softConsonantsU', 'test:softConsonantsL']
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', '@t:ЙЧЩ', 'NOCONTENT', 'SORTBY', 't')
+        env.assertEqual(res, expected, message=f'Dialect: {dialect}')
+
+        # Search hard consonants
+        expected = [2, 'test:hardConsonantsU', 'test:hardConsonantsL']
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', '@t:ЖШЦ', 'NOCONTENT', 'SORTBY', 't')
+        env.assertEqual(res, expected, message=f'Dialect: {dialect}')
+
+        # Search hard vowels
+        expected = [2, 'test:hardVowelsU', 'test:hardVowelsL']
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', '@t:АЭЫОУ', 'NOCONTENT', 'SORTBY', 't')
+        env.assertEqual(res, expected, message=f'Dialect: {dialect}')
+
+        # Search soft vowels
+        expected = [2, 'test:softVowelsU', 'test:softVowelsL']
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', '@t:ЯЕИЁЮ', 'NOCONTENT', 'SORTBY', 't')
+        env.assertEqual(res, expected, message=f'Dialect: {dialect}')
+
+def testDiacritics(env):
+    if not is_locale_available('en_US.UTF-8'):
+        env.skip()
+
+    conn = getConnectionByEnv(env)
+    env.cmd('FT.CREATE', 'idx', 'ON', 'HASH',
+            'LANGUAGE', 'ENGLISH', 'SCHEMA', 't', 'TEXT')
+
+    conn.execute_command('HSET', 'test:1', 't', 'éèêë')
+    conn.execute_command('HSET', 'test:2', 't', 'ÉÈÊË')
+    conn.execute_command('HSET', 'test:3', 't', 'àâä')
+    conn.execute_command('HSET', 'test:4', 't', 'ÀÂÄ')
+    conn.execute_command('HSET', 'test:5', 't', 'ç')
+    conn.execute_command('HSET', 'test:6', 't', 'Ç')
+    conn.execute_command('HSET', 'test:7', 't', 'œ')
+    conn.execute_command('HSET', 'test:8', 't', 'Œ')
+    conn.execute_command('HSET', 'test:9', 't', 'ùûü')
+    conn.execute_command('HSET', 'test:10', 't', 'ÙÛÜ')
+    conn.execute_command('HSET', 'test:11', 't', 'îï')
+    conn.execute_command('HSET', 'test:12', 't', 'ÎÏ')
+    conn.execute_command('HSET', 'test:13', 't', 'ôö')
+    conn.execute_command('HSET', 'test:14', 't', 'ÔÖ')
+    conn.execute_command('HSET', 'test:15', 't', 'ÿ')
+    conn.execute_command('HSET', 'test:16', 't', 'Ÿ')
+    conn.execute_command('HSET', 'test:17', 't', 'æ')
+    conn.execute_command('HSET', 'test:18', 't', 'Æ')
+
+    if not env.isCluster():
+        res = env.cmd(debug_cmd(), 'DUMP_TERMS', 'idx')
+        # only 9 terms are indexed, the lowercase representation of the terms
+        # with diacritics, but the diacritis are not removed.
+        env.assertEqual(len(res), 9)
+    
+def testDiacriticLimitation(env):
```

Comment:
Please add documentation as to what the test is testing (for the other tests as well). It's a nice TL;DR and in general can explain some "weird" stuff that happens in the test